src/seer_server.erl

```diff
@@ -0,0 +1,166 @@
+-module(seer_server).
+
+-include("seer.hrl").
+
+-behaviour(gen_server).
+
+-export([init/1, handle_cast/2, handle_call/3, handle_info/2, terminate/2, code_change/3]).
+-export([start_link/0]).
+
+-record(state,
+        {mode :: undefined | stdout | carbon | carbon_offline | carbon_reconnected,
```

Comment:
I have a feeling that the connection state to carbon should be encapsulated in a data type that includes the buffer, and should then be manipulated with some functions. Otherwise, this is looking more and more like a state machine, and not really in a good way.

That said, it's not essential. As long as the number of states doesn't keep growing, we could open a ticket for fixing this in the future. It might be worth putting off until we start talking about Prometheus.

Reply:
I completely agree. In my opinion, the next logical step in that project would be to refactor it to keep as much protocol-specific things outside of the main server, and allow for an easier switch between protocols. I would keep that out of the scope of the current release however.

Reply:
I'll create a Github issue for this. When we get around to supporting Prometheus, I think that would be the time to pay down this tech debt.